app/Singletons/AssetsManifest.php

```diff
@@ -12,14 +12,23 @@
 
 class AssetsManifest
 {
+    private const MANIFEST_TIMEOUT_INTERVAL_SECONDS = 3; // Interval at which to check for the manifest file if it's missing.
+    private const MANIFEST_TIMEOUT_SECONDS = 60; // Time limit to wait for a missing manifest file before throwing.
+
     private $manifest;
 
     public function __construct()
     {
         $manifestPath = public_path('assets/manifest.json');
+        $startTimeSeconds = microtime(true);
+
+        while (!file_exists($manifestPath)) {
```

Comment:
I wouldn't do it here but in `run.sh` instead as it shouldn't be needed for anything but docker dev (esp if something is broken on prod it'll probably never finish and just needlessly waiting).

At least this would need a logging somewhere because otherwise it will just hang with no indication of what's going on while waiting.

Reply:
alternatively, loading the manifest on first request is probably fine